subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -94,9 +105,20 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 				       buf, sizeof(buf))));
 		}
 
-		net_if_ipv6_addr_add(context->iface,
-				     (struct in6_addr *)(&address->mAddress),
-				     NET_ADDR_ANY, 0);
+		if_addr = net_if_ipv6_addr_add(
+					context->iface,
+					(struct in6_addr *)(&address->mAddress),
+					NET_ADDR_AUTOCONF, 0);
+
+		if (if_addr == NULL) {
+			NET_WARN("Not enough buffers to register OpenThread's"
+				 " IP address. Increase "
+				 "CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT");
```

Comment:
And I absolutely love self-diagnosing software. But as other things I love, I also like to know limits on them. For example, I do it like this (explicitly hint user that by enabling some config var they can get more info) in the shell. But that's because shell is a debugging feature. Speaking of *error messages*, this one would be first of such nature. (So, do we need such a long error message?)

Reply:
Sorry I don't follow this comment (beside of the last sentence).